trieste/acquisition/rule.py

```diff
@@ -1068,3 +1082,173 @@ def state_func(
             return state_, points
 
         return state_func
+
+
+class BatchHypervolumeSharpeRatioIndicator(
+    AcquisitionRule[TensorType, SearchSpaceType, ProbabilisticModelType]
+):
+    """Implements the Batch Hypervolume Sharpe-ratio indicator acquisition
+    rule, designed for large batches"""
+
+    def __init__(
+        self,
+        batch_size: int = 5,
+        ga_population_size: int = 500,
+        ga_n_generations: int = 200,
+        filter_threshold: float = 0.1,
+    ):
+        """
+        :param batch_size: The number of points in a batch. Defaults to 5.
+        :param ga_population_size: The population size used in the genetic algorithm
+             that finds points on the Pareto front. Defaults to 500.
+        :param ga_n_generations: The number of genenrations to run in the genetic
+             algorithm. Defaults to 200.
+        :param filter_threshold: The probability of improvement below which to exlude
+             points from the Sharpe ratio optimisation. Defaults to 0.1.
+        """
+        if not batch_size > 0:
+            raise ValueError(f"Batch size must be greater than 0, got {batch_size}")
+        if not ga_population_size >= batch_size:
+            raise ValueError(
+                "Population size must be greater or equal to batch size, got "
+                f"batch size as {batch_size} and population size as {ga_population_size}"
+            )
+        if not ga_n_generations > 0:
+            raise ValueError(f"Number of generation must be greater than 0, got {ga_n_generations}")
+        if not 0.0 <= filter_threshold < 1.0:
+            raise ValueError(f"Filter threshold must be in range [0.0,1.0), got {filter_threshold}")
+        if pymoo is None:
+            raise ImportError(
+                "BatchHypervolumeSharpeRatioIndicator requires pymoo, "
+                "which can be installed via `pip install trieste[qhsri]`"
+            )
+        builder = ProbabilityOfImprovement().using(OBJECTIVE)
+
+        self._builder: AcquisitionFunctionBuilder[ProbabilisticModel] = builder
+        self._batch_size: int = batch_size
+        self._population_size: int = ga_population_size
+        self._n_generations: int = ga_n_generations
+        self._filter_threshold: float = filter_threshold
+        self._acquisition_function: Optional[AcquisitionFunction] = None
+
+    def __repr__(self) -> str:
+        """"""
+        return f"""BatchHypervolumeSharpeRatioIndicator(
+        batch_size={self._batch_size}, ga_population_size={self._population_size},
+        ga_n_generations={self._n_generations}, filter_threshold={self._filter_threshold}
+        )
+        """
+
+    class MeanStdTradeoff(PymooProblem):  # type: ignore[misc]
```

Comment:
In general it's best to avoid inner classes, as they prevent objects from being pickled. You could instead make this a top level class but prefix its name with `_` (so `_MeanStdTradeoff`) to indicate that's it's private.